### PR TITLE
Create langtons ant algorithm

### PR DESCRIPTION
**Pull request by CaedenPH**
### Describe your change:
Implements the langtons ant algorithm
@ https://en.wikipedia.org/wiki/Langton%27s_ant

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: cellular_automata/langtons_ant.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    def __init__(self, width: int, height: int) -> None:
+        # Each square is either True or False where True is white and False is black
+        self.board = [[True] * width for _ in range(height)]
+        self.ant_position: tuple[int, int] = (width // 2, height // 2)
+
+        # Initially pointing left (similar to the the wikipedia image)
+        # (0 = 0° | 1 = 90° | 2 = 180 ° | 3 = 270°)
+        self.ant_direction: int = 3
+
+    def move_ant(self, axes: plt.Axes | None, display: bool, _frame: int) -> None:
```
Comment (tianyizheng02) [inline cellular_automata/langtons_ant.py:37]: What does `_frame` do? It doesn't appear to be used anywhere

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Overall looks good, only got some minor suggestions. Also, could you add a function to animate the ant? Currently the user must call `move_ant` and `display` repeatedly to see the ant move, and it'd be nice if that could be automated.

File: cellular_automata/langtons_ant.py (hunk 1 above)
Comment (CaedenPH) [reply cellular_automata/langtons_ant.py:37]: It doesn't do anything - It's an argument that is passed in from `FuncAnimation`
The preceding `_` indicates that its unused

File: cellular_automata/langtons_ant.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        >>> la.move_ant(None, True, 0)
+        >>> la.board
+        [[True, False], [True, False]]
+        """
+        directions = {
+            0: [-1, 0],  # 0°
+            1: [0, 1],  # 90°
+            2: [1, 0],  # 180°
+            3: [0, -1],  # 270°
+        }
```
[the remark this answers is not in the document]
Comment (CaedenPH) [reply cellular_automata/langtons_ant.py:62]: Good idea

**Comment by CaedenPH** [comment]
@tianyizheng02 
https://github.com/CaedenPH/Python/blob/langtons-ant/cellular_automata/langtons_ant.py#L95-L98
This animation is a [matplotlib animation](https://matplotlib.org/stable/api/_as_gen/matplotlib.animation.FuncAnimation.html) which calls `func` (`move_ant`) `frames` times. The default frames passed is `100_000` so it would animate langtons ant up to `100_000` iterations. Here is what it looks like

[screen-capture (13).webm](https://github.com/TheAlgorithms/Python/assets/23421705/26f2e30e-2ea6-4be1-a5ff-e168700f7dbe)

At the `11000`th iteration, this pattern is created
basic-ant

File: cellular_automata/langtons_ant.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        """
+        Displays the board without delay in a matplotlib plot
+        to visually understand and track the ant.
+
+        >>> _ = LangtonsAnt(WIDTH, HEIGHT)
+        """
+        fig, ax = plt.subplots()
+        # Assign animation to a variable to prevent it from getting garbage collected
+        self.animation = FuncAnimation(
+            fig, partial(self.move_ant, ax, True), frames=frames, interval=0
```
Comment (tianyizheng02) [inline cellular_automata/langtons_ant.py:96]: The interval should be set to a positive int. On my machine at least, the animation just doesn't run if the interval is 0

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

File: cellular_automata/langtons_ant.py (hunk 3 above)
Comment (CaedenPH) [reply cellular_automata/langtons_ant.py]: @tianyizheng02 Ah I see. Changed to `1`

**Review by tianyizheng02 — APPROVED** [review]

**Pull request merged**